src/ringct/rctSigs.cpp

```diff
@@ -118,19 +118,32 @@ namespace rct {
     }
     
     //see above.
-    bool verifyBorromean(const boroSig &bb, const key64 P1, const key64 P2) {
+    bool verifyBorromean(const boroSig &bb, const ge_p3 P1[64], const ge_p3 P2[64]) {
         key64 Lv1; key chash, LL;
         int ii = 0;
+        ge_p2 p2;
         for (ii = 0 ; ii < 64 ; ii++) {
-            addKeys2(LL, bb.s0[ii], bb.ee, P1[ii]);
+            // equivalent of: addKeys2(LL, bb.s0[ii], bb.ee, P1[ii]);
+            ge_double_scalarmult_base_vartime(&p2, bb.ee.bytes, &P1[ii], bb.s0[ii].bytes);
+            ge_tobytes(LL.bytes, &p2);
             chash = hash_to_scalar(LL);
-            addKeys2(Lv1[ii], bb.s1[ii], chash, P2[ii]);
+            // equivalent of: addKeys2(Lv1[ii], bb.s1[ii], chash, P2[ii]);
+            ge_double_scalarmult_base_vartime(&p2, chash.bytes, &P2[ii], bb.s1[ii].bytes);
+            ge_tobytes(Lv1[ii].bytes, &p2);
         }
         key eeComputed = hash_to_scalar(Lv1); //hash function fine
         return equalKeys(eeComputed, bb.ee);
     }
 
-    
+    bool verifyBorromean(const boroSig &bb, const key64 P1, const key64 P2) {
+      ge_p3 P1_p3[64], P2_p3[64];
```

Comment:
This pushes 20 KiB onto the stack. @hyc what do you think? Is this a concern for android builds, or any other platforms? Or does no one verify the blockchain on android?

Reply:
Given that we explicitly create all our threads with 5MB stacks I don't see much problem here.